operator/helper/multiline_test.go

```diff
@@ -164,8 +160,39 @@ func TestLineStartSplitFunc(t *testing.T) {
 				"LOGPART log1\nLOGPART log1",
 			},
 			Flusher: &Flusher{
-				force:           true,
-				lastForcedFlush: time.Now(),
+				// We assume than in previous iteration we had same data length
+				previousDataLength: len("LOGPART log1\nLOGPART log1\t   \n"),
+				lastDataChange:     time.Unix(0, 0),
+				forcePeriod:        time.Second,
+			},
+		},
+		{
+			Name:    "LogsWithFlusherWithMultipleLogsInBuffer",
+			Pattern: `^LOGSTART \d+`,
+			Raw:     []byte("LOGPART log1\nLOGSTART 123\nLOGPART log1\t   \n"),
+			ExpectedTokenized: []string{
+				// We expect all logs except last one, as it will be flushed in next iteration
+				"LOGPART log1",
```

Comment:
Shouldn't there be another token here?

Reply:
This is the perfect example for what I write before about strict unit tests being not so real life example.

This should be considered in file input operator point of view. File input operator reads data from file and run it against `splitFunc`. If there is some tokens to emit, it emits, otherwise it ends with some data unflushed. Filelog input operator is able to keep track pointer to last data being flushed. So in next iteration it takes data from that point (meantime can be more data added to the end of the file) and run against `splitFunc`. If there is no new data added to the end of the file for `forcePeriod`, flusher force to flush it.

This is hardly for me to explain, so I will add list just in case:

Filelog input operator:

1. Read file from previous pointer position (`0` for `start_at: beginning`, EOF for `start_at: end`, `pointer position` if already in loop
2. Run against `splitFunc` and update pointer position
  a. return tokens if any (will update pointer position to the end of last token)
  b. force flush if needed (no change since `forcePeriod`) (will update pointer position at the EOF)
  c. wait for more data (will do not update pointer position)
3. go to 1

---

So this test do not check `2b`, because in this unit case we can iterate only one, so there is no way to test `2a` and `2b` in current test and code design.

Code design limitation is that `2a` is considered as change and `lastDataChange` is updated to current timestamp, so `2b` will be skipped